### PR TITLE
Feature extract cspkg to allow Calamari conventions to be run

### PR DESCRIPTION
**Pull request by mjrichardson**

### CODE REVIEW

File: source/Calamari.Tests/Fixtures/Deployment/Azure/OctopusTestAzureSubscription.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using System.Security.Cryptography.X509Certificates;
+using Calamari.Deployment;
+using NUnit.Framework;
+using Octostache;
+
+namespace Calamari.Tests.Fixtures.Deployment.Azure
+{
+    public class OctopusTestAzureSubscription
+    {
+        public const string AzureSubscriptionId = "8affaa7d-3d74-427c-93c5-2d7f6a16e754";
```
Comment (Damovisa) [inline source/Calamari.Tests/Fixtures/Deployment/Azure/OctopusTestAzureSubscription.cs]: Do we have a subscription we're just going to leave up there for tests?

File: source/Calamari/Deployment/Conventions/ExtractAzureCloudServicePackageConvention.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                return;
+
+            Log.Verbose("Extracting cspkg");
+            var packagePath = deployment.Variables.Get(SpecialVariables.Action.Azure.CloudServicePackagePath);
+            using (var package = Package.Open(packagePath, FileMode.Open))
+            {
+                var manifest = AzureCloudServiceConventions.ReadPackageManifest(package);
+                var workingDirectory = deployment.CurrentDirectory;
+
+                ExtractContents(package, manifest, "ServiceDefinition", workingDirectory);
```
Comment (Damovisa) [inline source/Calamari/Deployment/Conventions/ExtractAzureCloudServicePackageConvention.cs]: Should these strings ("ServiceDefinition", "NamedStreams", etc.) be stored somewhere as known strings? They're used a couple of times...

File: source/Calamari/Integration/Azure/CloudServicePackage/ManifestSchema/PackageDefinition.cs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        public XElement ToXml()
+        {
+           return new XElement(ElementName,
+               MetaData.ToXml(), 
+               new XElement(PackageContentsElementName, Contents.Select(x => x.ToXml())),
+               new XElement(PackageLayoutsElementName, Layouts.Select(x => x.ToXml()))
+               ); 
+        }
+
+        public const string Schema = @"<?xml version=""1.0"" encoding=""utf-8""?>
```
Comment (Damovisa) [inline source/Calamari/Integration/Azure/CloudServicePackage/ManifestSchema/PackageDefinition.cs]: Embedded resource instead?

File: source/Calamari.Tests/Fixtures/Deployment/Azure/OctopusTestAzureSubscription.cs (hunk 1 above)
Comment (mjrichardson) [inline source/Calamari.Tests/Fixtures/Deployment/Azure/OctopusTestAzureSubscription.cs]: Yup.  And if the certificate is not installed on the machine, the test will be ignored.

File: source/Calamari/Deployment/Conventions/ExtractAzureCloudServicePackageConvention.cs (hunk 2 above)
Comment (mjrichardson) [inline source/Calamari/Deployment/Conventions/ExtractAzureCloudServicePackageConvention.cs]: Yes. Will do.

File: source/Calamari/Integration/Azure/CloudServicePackage/ManifestSchema/PackageDefinition.cs (hunk 3 above)
Comment (mjrichardson) [inline source/Calamari/Integration/Azure/CloudServicePackage/ManifestSchema/PackageDefinition.cs]: Agreed.

**Pull request merged**